addons/mrp_account/tests/test_analytic_account.py

```diff
@@ -432,3 +438,76 @@ def test_mo_qty_analytics(self):
         mo_form.save()
         self.assertEqual(mo.state, 'progress')
         self.assertEqual(self.analytic_account.balance, 0.0)
+
+    def test_mandatory_analytic_plan_production(self):
+        """
+        Tests that the distribution validation is correctly evaluated
+        The Production (Manufacturing Order) creation should be constrained by an analytic applicability rule if any relevant one.
+        """
+        manufacturing_order = self.env['mrp.production'].create({
+            'product_id': self.product.id,
+        })
+        self.assertTrue(manufacturing_order, "Manufacturing Order should be created without error")
```

Comment:
The error message does not provide any relevant info here, the code is clear enough
```suggestion
```
Same with almost all next assert actually (I would say that only the msg for the `assertRaises` might be useful)